src/fty_config_manager.cc

```diff
@@ -0,0 +1,448 @@
+                    log_debug("Configuration file name: %s", configurationFileName.c_str());
+                    cxxtools::SerializationInfo si;
+                    getConfigurationToJson(si, configurationFileName);
+                    configSiList[feature] = si;
+                }
+                // Set response
+                setSaveResponse(configSiList, respDto);
+                userData << respDto;
+            } 
+            else if (configQuery.action.compare(RESTORE_ACTION) == 0)
+            {
+                // To store response
+                dto::srr::SrrRestoreDtoList srrRestoreDtoList(STATUS_SUCCESS); 
+                // Get request and serialize it
+                cxxtools::SerializationInfo restoreSi;
+                JSON::readFromString(configQuery.data, restoreSi);
+                // Get data member
+                cxxtools::SerializationInfo siData = restoreSi.getMember(DATA_MEMBER);
+                cxxtools::SerializationInfo::Iterator it;
+                for (it = siData.begin(); it != siData.end(); ++it)
+                {
+                    dto::srr::SrrRestoreDto respDto(it->begin()->name(), STATUS_FAILED);
+                    // Build the augeas configuration file name.
+                    std::string configurationFileName = AUGEAS_FILES + m_parameters.at(respDto.name);
+                    log_debug("Restore configuration for: %s, with configuration file: %s", respDto.name.c_str(), configurationFileName.c_str());
+                    
+                    cxxtools::SerializationInfo siData = it->getMember(respDto.name).getMember(DATA_MEMBER);
+                    int returnValue = setConfiguration(&siData, configurationFileName);
+                    if (returnValue == 0)
+                    {
+                        log_debug("Restore configuration for: %s succeed!", respDto.name.c_str());
+                        respDto.status = STATUS_SUCCESS;
+                    } 
+                    else
+                    {
+                        std::string errorMsg = "Restore configuration for: " + respDto.name + " failed!";
+                        log_error(errorMsg.c_str());
+                        respDto.error= errorMsg;
+                        srrRestoreDtoList.status = STATUS_FAILED;
+                    }
+                    srrRestoreDtoList.responseList.push_back(respDto);
+               }
+               userData << srrRestoreDtoList;
+            } 
+            else
+            {
+                throw ConfigurationException("Wrong command");
+            }
+            // Send response
+            sendResponse(msg, userData, configQuery.action);
+        } catch (const std::out_of_range& oor)
+        {
+            log_error("Feature name not found");
+        } catch (std::exception &e)
+        {
+            log_error("Unexpected error: %s", e.what());
+        } catch (...)
+        {
+            log_error("Unexpected error: unknown");
+        }
+    }
```

Comment:
We should probably modify fty-common-messagebus to wrap callback calls inside a try/catch instead of requiring clients to do that themselves.

Reply:
I disagree. The client should decide his strategy regarding error.

Reply:
We could recommend it, if he wants an approach type "show must go one"